unit_test/batched/Test_Batched_TeamVectorSolveUTV2.hpp

```diff
@@ -0,0 +1,250 @@
+
+      auto ww = Kokkos::subview(_w, k, Kokkos::ALL());
+
+      // make diagonal dominant and set xx = 1,2,3,4,5
+      const int m = aa.extent(0), r = rr.extent(1);
+      if (m <= r) {
+	Kokkos::parallel_for
+	  (Kokkos::TeamVectorRange(member, m),
+	   [&](const int &i) {
+	     aa(i,i) += add_this;
+	     for (int j=0;j<2;++j)
+	       xx(i,j) = (i+1);
+	   });
+      } else {
+	Kokkos::parallel_for
+	  (Kokkos::TeamVectorRange(member, m*m),
+	   [&](const int &ij) {
+            const int i = ij/m, j = ij%m;
+            value_type tmp(0);
+            for (int l=0;l<r;++l)
+              tmp += rr(i,l)*rr(j,l);
+            aa(i,j) = tmp;
+          });
+	Kokkos::parallel_for
+	  (Kokkos::TeamVectorRange(member, m),
+	   [&](const int &i) {
+	     for (int j=0;j<2;++j)
+	       xx(i,j) = (i+1);
+          });
+      }
+      
+      /// copy for verification
+      TeamVectorCopy<MemberType,Trans::NoTranspose>
+	::invoke(member, aa, ac);
+
+      /// bb = AA*xx
+      TeamVectorGemm<MemberType,Trans::NoTranspose,Trans::NoTranspose,Algo::Gemm::Unblocked>
+	::invoke(member, one, aa, xx, zero, bb); 
+      member.team_barrier();
+      
+      /// Solving Ax = b using UTV transformation
+      /// A P^T P x = b
+      /// UTV P x = b;
+
+      /// UTV = A P^T
+      int matrix_rank(0);
+      TeamVectorUTV<MemberType,AlgoTagType>
+	::invoke(member, aa, pp, uu, vv, ww, matrix_rank);
+      member.team_barrier();
+
+      TeamVectorSolveUTV<MemberType,AlgoTagType>
+	::invoke(member, matrix_rank, uu, aa, vv, pp, xx, bb, ww);
+    }
+
+    inline
+    void run() {
+      typedef typename MatrixViewType::non_const_value_type value_type;
+      std::string name_region("KokkosBatched::Test::TeamVectorSolveUTV");
+      std::string name_value_type = ( std::is_same<value_type,float>::value ? "::Float" : 
+                                      std::is_same<value_type,double>::value ? "::Double" :
+                                      std::is_same<value_type,Kokkos::complex<float> >::value ? "::ComplexFloat" :
+                                      std::is_same<value_type,Kokkos::complex<double> >::value ? "::ComplexDouble" : "::UnknownValueType" );                               
+      std::string name = name_region + name_value_type;
+      Kokkos::Profiling::pushRegion( name.c_str() );
+
+      const int league_size = _a.extent(0);
+      Kokkos::TeamPolicy<DeviceType> policy(league_size, Kokkos::AUTO);
+
+      Kokkos::parallel_for(name.c_str(), policy, *this);
+      Kokkos::Profiling::popRegion(); 
+    }
+  };
+
+  template<typename DeviceType,
+           typename MatrixViewType,
+           typename VectorViewType,
+	   typename PivViewType,
+           typename WorkViewType,
+           typename AlgoTagType>
+  void impl_test_batched_solve_utv2(const int N, const int BlkSize) {
+    typedef typename MatrixViewType::non_const_value_type value_type;
+    typedef Kokkos::Details::ArithTraits<value_type> ats;
+    //const value_type one(1);
```

Comment:
Minor format comment, this line can be removed; not necessary to change in the PR, just a note to cleanup later.